File: Leet 3020/solution.cs
public class Solution
{
    public int MaximumLength(int[] nums)
    {
        Dictionary<long, int> frequency = [];
        foreach(int num in nums)
        {
            frequency.TryGetValue(num, out int val);
            frequency[num] = val + 1;
        }

        frequency.TryGetValue(1, out int result);
        if (result > 0)
        {
            frequency[1] = 0;
        }

        if (result % 2 == 0)
        {
            result--;
        }

        foreach(var key in frequency.Keys)
        {
            int count = 0;
            long curr = key;

            while (frequency.TryGetValue(curr, out int val) && val > 1)
            {
                count += 2;
                curr *= curr;
            }
            
            int add = frequency.ContainsKey(curr) ? 1 : -1;
            result = Math.Max(result, count + add);
        }

        return result;
    }
}

class Program
{
    private static void Main()
    {
        Console.WriteLine("3020. Find the Maximum Number of Elements in Subset");

        Solution solution = new();

        int[] nums1 = [5,4,1,2,2];
        Console.WriteLine($"Result: {solution.MaximumLength(nums1)}");
        
        int[] nums2 = [1,3,2,4];
        Console.WriteLine($"Result: {solution.MaximumLength(nums2)}");

    }
}

File: Leet 3020/solution.py
def maximumLength(nums: list[int]) -> int:
    freq: dict[int, int] = {}
    for num in nums:
        freq[num] = freq.get(num, 0) + 1

    result: int = freq.get(1, 0)
    if result > 0:
        freq[1] = 0

    if result % 2 == 0:
        result -= 1

    for key in freq:
        count: int = 0
        curr: int = key
        while freq.get(curr, 0) > 1:
            count += 2
            curr *= curr

        result = max(result, count + (1 if curr in freq else -1))
        
    return result
    

def main() -> None:
    print("3020. Find the Maximum Number of Elements in Subset")
    print(f"Result: {maximumLength([5,4,1,2,2])}")


if __name__ == "__main__":
    main()